lib/Sema/SemaInit.cpp

```diff
@@ -7093,6 +7093,15 @@ InitializationSequence::Perform(Sema &S,
       CurInit = CurInitExprRes;
 
       bool Complained;
+
+      // FIXME : This code is inserted only as a hotfix for type variable
```

Comment:
I think the comment should be different.    This is a workaround because generic type instantiation has not been implemented.  That leaves type variables in unexpected places.    You've actually implemented compatibility for type variables in the current change.  Also, could you use the word TODO instead of FIXME.   FIXME implies that this is a bug, whereas TODO means unimplemented functionality.